designs/RestoreAndBuildCompatibility.md

```diff
@@ -156,23 +175,38 @@ Given that the implied compatibility exists, the proposal is for the SDK build t
 * Warn when the restore output is from an older tool than anticipated.
 * Warn when the restore output is from a newer tool than anticipated.
 
-There are 3 technical options and each might have a different effect on the frequency in which this warning is raised.
-
-#### Option 1 - Use assets file version
+### Assets file version
 
 NuGet increments the `project.assets.json` version every time there's a functional change in behavior. The build tasks in the SDK are built with a specific version of NuGet, and the SDK could raise a warning when the `project.assets.json` version is not the one it expects.
 
-Cons:
+This version that has historically been used for schema changes, not tooling compatibility checks. For example, aliases addition involved adding more information to the assets file, but not a schema change so this number was not incremented.

```

Comment:
Adding new properties into a JSON structure is a schema change, just as adding a new column to a SQL database table is a schema change. The change might be backwards compatible for most customers, and by not incrementing the schema version number, customers affected customers will get a compile error that doesn't make a lot of sense, rather than a clear "use a newer version of nuget" message.

For examples like package/assembly aliases, I think we do need to increment the schema number, but if we're concerned about forcing customers using old nuget.exe having to upgrade when it would otherwise work, then I think NuGet needs to do "feature detection" and when the feature (alias in this example) is not used, write the lower version number, and when the feature is used, write the higher version number. The .NET SDK will need to support both version numbers. But this is the only way I can see customers getting a good, actionable error message, rather than a bizzare compile time error.

However, the effort and complexity of implementing this may not be worthwhile compared to a strong push to get customers to move to msbuild and dotnet cli restore.

Reply:
There are ways to do it without a schema change. But yeah the `nuances` of the `break` cause trouble to say the least.

Reply:
My previous comment was a long way to say that I disagree with the markdown file when it says "not a schema change so the number was not incremented".  It was a schema change, but we didn't increment the number. It's misrepresenting (in my opinion) what actually happened and the reason.